src/common/http_defs.cc

```diff
@@ -62,6 +62,17 @@ namespace Pistache::Http
             return !in.fail();
         }
 
+        bool parse_epoch(const std::string& s, time_point& tp)
+        {
+            for (unsigned int i = 0; i < s.size(); ++i)
+            {
+                if (!std::isdigit(s[i]))
+                    return false;
+            }
+            tp = time_point(std::chrono::seconds(std::stoull(s)));
```

Comment:
Is this sufficient to validate the input string? Can `time_point` signal to the caller in any way an error?

Reply:
Oh yeah.

Well, `std::stoull` can have exceptions `std::invalid_argument` or `std::out_of_range.` I guess we should trap an exception, set `errno`, and return `false`. 

I don't see exceptions listed for `duration` or `time_point`, but maybe they could (e.g. overflow?). Probably safest would be to _try/catch_ round that line, set `errno` on `out_of_range (ERANGE)` or `EINVALID` on any other exception, and return false in the exception case.

Reply:
true, we could try catch there. `invalid_argument` won't be thrown as it is all digits, but out of range is a fair possibility.

Reply:
I've added the try catch.